### DISCUSSION TITLE
automatic uploads not working

### DISCUSSION THREAD

**Discussion by braddirt** [earlephilhower/arduino-pico#558] [General]
I recently ran into an issue where I stopped being able to have Arduino IDE automatically upload the programs to the Pico. I hit upload after making sure to select the right board and port. Then it compiles and gives this error.

>  Sketch uses 60088 bytes (2%) of program storage space. Maximum is 2093056 bytes.
> Global variables use 11216 bytes (4%) of dynamic memory, leaving 250928 bytes for local variables. Maximum is 262144 bytes.
> Resetting COM4
> Converting to uf2, output size: 133632, start address: 0x2000
> Unable to build drive list
> An error occurred while uploading the sketch

The board resets and opens up as a storage device during this process
I am then able to locate the uf2 file that Arduino stored in my temp files and drag that onto the Pico for programming but it refuses to do that automatically. I have tried to uninstall and reinstall Arduino IDE, The board cores, and the device drivers to get this to work again. I even tried removing the Arduino15 folder. Does anyone have an idea of what it might be or what to try next?

Edition	Windows 10 Home
Version	21H1
Installed on	‎8/‎29/‎2021
OS build	19043.1645
Experience	Windows Feature Experience Pack 120.2212.4170.0
Arduino 1.8.19

**Comment by earlephilhower** [comment]
What version of the core is installed?  In 1.11.0 there was a patch to work around some MSFT changes to system apps which might be what you're seeing if you're on an older version.  If you're at earlier than 1.11.0, upgrade and give it another try.

**Reply by braddirt** [reply]
Well, I guess I got the brand new one in my latest reinstall process, 2.0.0. It is still having those issues.

**Reply by pinchies** [reply]
@braddirt This may also help you -- this error can be caused by a corrupted windows PATH environment variable. Please see my blog post here about the ["Unable to build drive list" error when uploading](https://samuelpinches.com.au/hacking/problem-uploading-sketch-to-raspberry-pico-rp2040-in-arduino-ide-unable-to-build-drive-list/).
